module/jobqueue/component_consumer_test.go

```diff
@@ -182,6 +184,7 @@ func (suite *ComponentConsumerSuite) TestProgressesOnComplete() {
 	jobData := generateTestData(testJobsCount)
 	finishedJobs := make(map[uint64]bool, testJobsCount)
 
+	mu := sync.Mutex{}
```

Comment:
This mutex is pointless since you're no guarding your read with it ... this is still not threadsafe.

Reply:
read is synchronized by the `done` channel beloww

Reply:
you're relying on an accidentally memory fence that only happens on test passes. when test fail, you're reading unguarded data

Reply:
The test will stop and hence not progress beyond this fence.  That said you have a point: explicit guard is better, let me add it.